From 4c77fe86441dda5753e8584af92010f362f83155 Mon Sep 17 00:00:00 2001
From: Joel Lehtonen <joel.lehtonen@iki.fi>
Date: Sun, 31 Oct 2010 03:32:09 +0300
Subject: [PATCH] ep93xx: Support for Codegrove TCU 0.1

Added support for W1 bus on Codegrove TCU 0.1.

Conflicts:

	arch/arm/mach-ep93xx/ts72xx.c
---
 arch/arm/mach-ep93xx/ts72xx.c |   18 ++++++++++++++++++
 1 files changed, 18 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-ep93xx/ts72xx.c b/arch/arm/mach-ep93xx/ts72xx.c
index bf3a666..505292c 100644
--- a/arch/arm/mach-ep93xx/ts72xx.c
+++ b/arch/arm/mach-ep93xx/ts72xx.c
@@ -26,6 +26,7 @@
 #include <linux/spi/spi.h>
 
 #include <mach/ep93xx_spi.h>
+#include <linux/w1-gpio.h>
 #include <mach/hardware.h>
 #include <mach/ts72xx.h>
 
@@ -369,6 +370,20 @@ static struct i2c_gpio_platform_data ts72xx_i2c_gpio_data = {
 static struct i2c_board_info __initdata ts72xx_i2c_board_info[] = {
 };
 
+/* W1 on Codegrove boards */
+
+static struct w1_gpio_platform_data ahma_w1_gpio_pdata = {
+        .pin            = EP93XX_GPIO_LINE_EGPIO8,
+        .is_open_drain  = 1,
+};
+
+static struct platform_device ahma_w1_device = {
+        .name                   = "w1-gpio",
+        .id                     = -1,
+        .dev.platform_data      = &ahma_w1_gpio_pdata,
+};
+
+
 /*************************************************************************
  * SPI
  *************************************************************************/
@@ -447,6 +462,9 @@ static void __init ts72xx_init_machine(void)
 
 	/* PWM1 is DIO_6 on TS-72xx header */
 	ep93xx_register_pwm(0, 1);
+	
+	/* W1 on Codegrove boards */
+	platform_device_register(&ahma_w1_device);
 }
 
 MACHINE_START(TS72XX, "Technologic Systems TS-72xx SBC")
-- 
1.7.2.5

